#!/usr/bin/env python
PACKAGE = "navex_navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("global_frame_id", str_t, 0, "global static frame id, like a /world or /map", "/map")
gen.add("scale_linear", double_t, 0, "Linear speed scale factor", 1.0, -5.0, 5.0)
gen.add("scale_angular", double_t, 0, "Angular speed scale factor", 1.0, -5.0, 5.0)

gen.add("goal_tolerance", double_t, 0, "Goal tolerance radius in meters", 0.2, 0.01, 10.0)

gen.add("replanning_frequency", double_t, 0, "requency of unconditional replanning [Hz]", 0.0, 0, 10);


# gen.add("int_param",    int_t,    0, "An Integer parameter", 50,  0, 100)
# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)

# size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                        gen.const("Medium",     int_t, 1, "A medium constant"),
#                        gen.const("Large",      int_t, 2, "A large constant"),
#                        gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
#                      "An enum to set size")

# gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "navex_navigation", "NavigationParameters"))